hadoop-tools/hadoop-distcp/src/main/java/org/apache/hadoop/tools/CopyListing.java

```diff
@@ -305,6 +305,33 @@ public static CopyListing getCopyListing(Configuration configuration,
     }
   }
 
+  /**
+   * Public Factory method with which the appropriate Diff CopyListing implementation may be retrieved.
+   * @param configuration The input configuration.
+   * @param credentials Credentials object on which the FS delegation tokens are cached
+   * @param distCpSync DistcpSync object used to sync diffs between source and target.
+   * @return An instance of the appropriate CopyListing implementation.
+   * @throws java.io.IOException - Exception if any
+   */
+  public static CopyListing getDiffCopyListing(Configuration configuration,
+      Credentials credentials, DistCpSync distCpSync) throws IOException {
+    String copyListingClassName =
```

Comment:
Why are we assigning default from config here? you are always reassiging with copyListingClass.getName(); right?

Reply:
Removed this now